packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/ecs/run-task.ts

```diff
@@ -436,26 +455,40 @@ export class EcsRunTask extends sfn.TaskStateBase implements ec2.IConnectable {
   }
 }
 
-function renderOverrides(containerOverrides?: ContainerOverride[]) {
-  if (!containerOverrides || containerOverrides.length === 0) {
+interface OverrideProps {
+  cpu?: string;
+  memoryMiB?: string;
+  containerOverrides?: ContainerOverride[];
+}
+
+function renderOverrides(props: OverrideProps) {
+  const containerOverrides = props.containerOverrides;
+  const noContainerOverrides = !containerOverrides || containerOverrides.length === 0;
+  if (noContainerOverrides && !props.cpu && !props.memoryMiB) {
     return undefined;
   }
 
   const ret = new Array<any>();
-  for (const override of containerOverrides) {
-    ret.push({
-      Name: override.containerDefinition.containerName,
-      Command: override.command,
-      Cpu: override.cpu,
-      Memory: override.memoryLimit,
-      MemoryReservation: override.memoryReservation,
-      Environment:
-        override.environment?.map((e) => ({
-          Name: e.name,
-          Value: e.value,
-        })),
-    });
+  if (!noContainerOverrides) {
```

Comment:
can we keep this same as before `!containerOverrides || containerOverrides.length === 0`,  it was more readable than current negation of negation.

Reply:
@shikha372 
I am sorry. However, it is a complex branch and we do not want to clone the same expression. Any suggestions on how to write better code would be appreciated.

Reply:
Thanks @kackyt , looked at it my initial thought was it would be `!containerOverrides || containerOverrides.length === 0` but seems like not, it would be something like `if (containerOverrides || containerOverrides.length !== 0)` in which case its a negation anyways, i would go with the current one.